featuretools/feature_discovery/LiteFeature.py

```diff
@@ -0,0 +1,260 @@
+            df_id=self.df_id,
+            idx=self.idx,
+        )
+
+    def get_primitive_name(self) -> Union[str, None]:
+        return self.primitive.name if self.primitive else None
+
+    def get_dependencies(self, deep=False) -> List[LiteFeature]:
+        flattened_dependencies = []
+        for f in self.base_features:
+            flattened_dependencies.append(f)
+
+            if deep:
+                dependencies = f.get_dependencies()
+                if isinstance(dependencies, list):
+                    flattened_dependencies.extend(dependencies)
+                else:
+                    flattened_dependencies.append(dependencies)
+        return flattened_dependencies
+
+    def get_origin_features(self) -> List[LiteFeature]:
+        all_dependencies = self.get_dependencies(deep=True)
+        return [f for f in all_dependencies if f.depth == 0]
+
+    def __repr__(self) -> str:
+        base_features = ", ".join([f"{x.id[:5]}..." for x in self.base_features])
+        return f"LiteFeature(name='{self.get_name()}', logical_type={self.logical_type}, tags={self.tags}, primitive={self.get_primitive_name()}, base_features=[{base_features}], df_id=None, id='{self.id[:5]}...' n_output_features={self.n_output_features} idx={self.idx})"
+
+    @property
+    def column_schema(self) -> ColumnSchema:
+        return ColumnSchema(logical_type=self.logical_type, semantic_tags=self.tags)
+
+    def rename(self, name: str):
+        self.name = name
+
+    def get_name(self) -> str:
+        if self.name:
+            return self.name
+        elif len(self.related_features) > 0:
+            return f"{self._gen_name}[{self.idx}]"
+        return self._gen_name
+
+    def get_depth(self) -> int:
+        return self.depth
+
+    def dependendent_primitives(self) -> Set[Type[PrimitiveBase]]:
+        dependent_features = self.get_dependencies(deep=True)
+        dependent_primitives = {
+            type(f.primitive) for f in dependent_features if f.primitive
+        }
+        if self.primitive:
+            dependent_primitives.add(type(self.primitive))
+        return dependent_primitives
+
+    def to_dict(self) -> Dict[str, Any]:
+        return {
+            "name": self.name,
+            "logical_type": self.logical_type.__name__ if self.logical_type else None,
+            "tags": list(self.tags),
+            "primitive": hash_primitive(self.primitive)[0] if self.primitive else None,
+            "base_features": [x.id for x in self.base_features],
+            "df_id": self.df_id,
+            "id": self.id,
+            "related_features": [x.id for x in self.related_features],
+            "idx": self.idx,
+        }
+
+    def is_multioutput(self) -> bool:
+        return len(self.related_features) > 0
+
+    def copy(self) -> LiteFeature:
+        copied_feature = LiteFeature(
+            name=self.name,
+            logical_type=self.logical_type,
+            tags=self.tags,
```

Comment:
Is this still an issue? Not exactly sure what you meant.

Reply:
Ok so the vast majority of features will hit `self._tags = self._tags | self._logical_type.standard_tags` which makes a new set object, but in this very specific case, it doesn't hit that line and the tags object is the same 

```python
    f1 = LiteFeature("f1", Double)
    f2 = LiteFeature("f2", Double)

    f3 = LiteFeature(
        name="Column 1",
        base_features=[f2, f1],
        logical_type=Double,
        tags={"index"}
    )

    f_copy = f3.copy()

    f_copy.tags.add("hi")
    assert "hi" in f3._tags
```

I think that copying `tags` at the property would solve this case (assuming people don't go through `_tags`) but I think it's safest to copy any mutable objects when constructing the copied feature

Reply:
Though if LiteFeatures are truly immutable, I think you're right that this isn't something to worry about since the fear would be that mutating the copied feature would mutate the original feature.